src/latexeditorview.cpp

```diff
@@ -536,6 +539,44 @@ bool DefaultInputBinding::contextMenuEvent(QContextMenuEvent *event, QEditor *ed
 			edView->connect(act, SIGNAL(triggered()), edView, SLOT(triggeredThesaurus()));
 			contextMenu->addAction(act);
 		}
+		{
+			//	copy managed action "previewLatex" from texstudio.cpp
+//			QAction *act = new QAction(LatexEditorView::tr("Pre&view Selection/Parentheses"));
+			QAction *act = new QAction();
```

Comment:
I am really skeptical about this copying actions stuff.
Basically you have 2 actions with the same short cut, not sure that this leads to well defined behavior when one of them is discarded. 

And again, code duplication ... 3 nearly identical for loops...

Reply:
probably the cleanest solution for this is, that the context menu point is stored in DefaultInput... as extra variable and declared invalid/deleted when the context menu is closed.
texstudio checks in edView if the point is present and act accordingly.
This would avoid any hacks on the actions.

Reply:
I can't use shortcuts while the (contex) menu is open. Especially pressing the Alt-key immediately closes the (context) menu. So I conclude that the (context) menu shows the shortcut merely as information. If you remove the shortcut from the main menu action (config dialog) it will no longer be present in the context menu.

Do we know that the context menu is present as long as the triggered action is running?

Thesaurus, as an example, is in the main menu with shortcut but there is a new action created in latexeditorview with no shortcut. The new action is used to pass coordinate information to the final function. But the preview actions are in baseActions. These are added as is (in the main menu) to the context menu. So they show the same (probably translated) text and the same shortcut.  "Copying" them into new actions repeats the procedure used for Thesaurus and avoids repeating translations and shortcuts, but allows for new slots as with Thesaurus.

I didn't manage to write a function for the loops, sorry.

Thus I would prefer to stay with this solution. Could you provide a case that you assume to have issues with the solution?

Thx